Remove the separate SecureBoot USB key

kerneis-anssi opened the pull request:
Remove deprecated scripts (after merging a few relevant changes) and update documentation accordingly.

kerneis-anssi committed:
Add vim, kbd and console-data packages

- vim is useful to fix scripts
- kbd and console-data make it possible to use `loadkeys fr` to change
  the keyboard layout in the console.

kerneis-anssi committed:
Avoid code duplication for cleanup

kerneis-anssi committed:
Add requirement checking to create-chipsec.sh

Copied from create-shell.sh in order to deprecate it.

kerneis-anssi committed:
Remove the separate EFI Shell bootable key

kerneis-anssi committed:
Regenerate documentation

kerneis-anssi committed:
Rename French README to give a markdown extension

kerneis-anssi committed:
Address shellcheck warnings

kerneis-anssi committed:
Fix path to kernel module signing script

yaperez-anssi commented on tools/create-chipsec.sh:
```diff
@@ -165,8 +165,12 @@ install_debian () {
 	do_chroot apt -y install git build-essential linux-headers-amd64
 	do_chroot apt -y install python3 python3-dev python3-setuptools
-	do_chroot apt -y install sed nasm pciutils fwupd lshw usbutils
+	do_chroot apt -y install vim sed nasm pciutils fwupd lshw usbutils
```
No strong objection from me, but I usually manually used EXTRA_PACKAGES for that in order to keep the final media small enough (when building the ISO format the 2G size is to fit the maximum file size on Github, and it's nice to have some spare room on the root fs)

kerneis-anssi replied:
There is ~300MiB left in the partition. The packages we're talking about are a few megabytes large. I'll also get rid of git to reclaim some space.

yaperez-anssi commented on tools/create-chipsec.sh:
```diff
@@ -165,8 +165,12 @@ install_debian () {
+	# Skip dpkg question about console-data configuration, we only care
+	# about having a database for loadkeys.
+	do_chroot sh -c "echo 'console-common	console-data/keymap/policy	select	Keep kernel keymap' | debconf-set-selections"
+	do_chroot apt -y install kbd console-data
```
Likewise, _loadkeys_ is useful when the default (qwerty) keymap is not suitable. I'm not sure we really need to be able to change the keymap dynamically. If it's possible for one tester to generate a key with an unique keymap (suitable for her) it might be better to do that.

But in the end, if kbd, console-data and dependencies are small enough, it's a bit of a moot point.

Again, no strong objection from me.

yaperez-anssi commented on tools/create-chipsec.sh:
```diff
@@ -62,7 +62,7 @@ sign_kernel () {
 
 	# Also sign the Chipsec module
-	"${mount_point}"/usr/lib/linux-kbuild-4.19/scripts/sign-file \
+	"${mount_point}"/usr/lib/linux-kbuild-*/scripts/sign-file \
```
As far as I remember, the linux-kbuild path doesn't change often (and especially not at each runtime kernel version), and it might be possible to have two installed at the same time (right now in unstable there are both `-5.10` and `-5.13`. Not sure if that can happens as part of the debootstrap process (maybe unlikely) but it's worth checking.

kerneis-anssi replied:
So the reason I changed it is because it broke with recent updates. Fair point about the possibility of several kernels (which I addressed somewhere else in this file). It doesn't seem to happen in practice in debootstrap, but fixed anyway.

yaperez-anssi commented on doc/build_chipsec.md:
```diff
@@ -14,60 +15,49 @@ A Help to build your own ChipSec and SecureBoot USB keys
+1. boot on the USB key, then at the bootloader prompt start "Debian GUN/Linux"
 1. when finished booting, login as root (no password)
+1. if you need an alternate keyboard, use eg. `loadkeys fr`.
 1. from the root terminal, launch ChipSec with "chipsec_main.py".
```
Nitpick but maybe use backticks for script filenames?

yaperez-anssi commented on doc/build_chipsec.md:
```diff
@@ -14,60 +15,49 @@ A Help to build your own ChipSec and SecureBoot USB keys
+   with commmands "fs0:" or "fs1:" or fsX: ... then "dir". Within Keytool, the
+   drives will have names instead. The USB key should be named "ESP", but you
+   can confirm it by browsing its content through the "Execute Binary" menu.
+1. With Keytool, replace (in that order) db, KEK and PK using files from "EFI/keys" folder
```
Since fc70cc9 an empty DBX (denylist) is generated as well and thus should be added as well. Also it might be worth noting to use `DBX.esl`, `DB.esl`, `KEK.esl` **but** `PK.auth`.

kerneis-anssi replied:
Done.

yaperez-anssi reviewed (commented):
Thanks! It looks good to me. There are a few very minor nitpicks but after addressing (or commenting on) them, it's ok to merge.

kerneis-anssi committed:
Do not install git on the USB key

Using git from host allows cloning a chipsec fork stored on the local
filesystem (with `-r /path/to/chipsec`).

kerneis-anssi committed:
Various fixes about formatting

- formatting on a USB key sometimes fails; printing the partition table
  adds a slight delay which seems to make it work more reliably.
- 1.6GiB is parsed as 6GiB by sfdisk; use 1600MiB instead.
- ensure FAT32 for the data partition.
